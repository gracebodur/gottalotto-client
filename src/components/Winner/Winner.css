section {
    display: flex;
    flex-direction: row;
    padding: 15px;
}

.Winner {
    text-align: center;
    display: flex;
    justify-content: center;
    color: #FFF;
    font-size: 30px;
}
