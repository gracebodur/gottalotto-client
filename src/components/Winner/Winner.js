import React, { Component } from 'react'
import avatar from '../../img/winner_avatar.png'
import Flippy, { FrontSide, BackSide } from 'react-flippy'
import './Winner.css'

class Winner extends Component {
	render() {
		const { user_name, guess_1, guess_2, guess_3, guess_4, guess_5, power_ball, message, guess_created_date, has_won } = this.props.winner
		return (
			<section className='Winner'>
			 <div style={{ display: 'flex', flex: '1 0 200px', justifyContent: 'space-around', 'flexWrap': 'wrap' }}>
			<Flippy
				flipOnHover={ true }
				flipDirection='vertical'
				style={{ width: '300px', height: '400px' }} 
			>
			<FrontSide style={{ 
				backgroundColor: '#41669d',
				display: 'flex',
        		alignItems: 'center',
        		flexDirection: 'column'}} >
				{/* <img src={avatar} alt='avatar' /> */}
					{user_name}
				<span 
				className='winnerName'
        		style={{
          		fontSize:'12px',
          		position: 'absolute',
          		bottom: '10px',
          		width: '100%'
        		}}>
      			</span>
			</FrontSide>
			<BackSide style={{ 
				backgroundColor: '#175852', 
				display: 'flex',
        		alignItems: 'center',
        		justifyContent: 'center',
        		flexDirection: 'column'}}>
					{message}
				<span className='winnerNumbers'
        		style={{
          		fontSize:'18px',
          		position: 'absolute',
          		bottom: '10px',
          		width: '100%'
        		}}>{guess_1}, {guess_2}, {guess_3}, {guess_4}, {guess_5}, PB: {power_ball}<br />
				</span>
			</BackSide>	
			</Flippy>
			</div>
			</section>
		)
	}
}

export default Winner
