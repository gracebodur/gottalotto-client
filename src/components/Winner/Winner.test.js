import React from "react";
import toJson from "enzyme-to-json";
import Winner from "./Winner";

describe(`Winner component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a Winner by default", () => {
		const wrapper = <Winner />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the Winner given props", () => {
		const wrapper = <Winner {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
