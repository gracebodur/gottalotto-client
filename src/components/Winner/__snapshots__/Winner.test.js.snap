// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner component renders a Winner by default 1`] = `undefined`;

exports[`Winner component renders the Winner given props 1`] = `undefined`;
